OpenRA.Mods.Common/Traits/World/VariedCostManager.cs

```diff
@@ -0,0 +1,69 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2020 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Attach this to the world actor.")]
+	public class VariedCostManagerInfo : TraitInfo
+	{
+		[Desc("Interval between new pricings measured in ticks.")]
+		public readonly int AdjustmentDelay = 1000;
+
+		[Desc("Range of percentage modifiers to apply.")]
+		public readonly int2 Multiplier = new int2(75, 100);
+
+		public override object Create(ActorInitializer init) { return new VariedCostManager(this, init.Self); }
+	}
+
+	public class VariedCostManager : ITick
+	{
+		public Dictionary<ActorInfo, int> CachedCostPercentage = new Dictionary<ActorInfo, int>();
+
+		readonly VariedCostManagerInfo info;
+		readonly IEnumerable<ActorInfo> actorInfos;
+
+		int tick;
+		Dictionary<string, int> groupPricing = new Dictionary<string, int>();
```

Comment:
This can be made `readonly`.

Reply:
I still seem to have not understood when things can be `readonly` but VS Code also suggests it so here we go.